nni/algorithms/compression/pytorch/quantization/quantizers.py

```diff
@@ -120,6 +123,222 @@ def quant_backward(tensor, grad_output, quant_type, scale, zero_point, qmin, qma
         return grad_output
 
 
+class ObserverQuantizer(Quantizer):
+    """This quantizer uses observers to record weight/activation statistics to get quantization information.
+    The whole process can be divided into three steps:
+        1. It will register observers to the place where quantization would happen (just like registering hooks).
+        2. The observers would record tensors' statistics during calibration.
+        3. Scale & zero point would be obtained after calibration.
+    Note that the observer type, tensor dtype and quantization qscheme are hard coded for now. Their customization
+    are under development and will be ready soon.
+    """
+
+    def __init__(self, model, config_list, optimizer=None):
+        super().__init__(model, config_list, optimizer)
+        # NOTE: this quantizer is experimental for now. The dtype and qscheme of quantization
+        # is hard-coded.
+        # TODO:
+        # 1. support dtype and qscheme customization through config_list. Current settings:
```

Comment:
if dtype and qscheme can be applied on each layer separately, then it is better to support them in config_list. otherwise, it is better to support them as quantizer's initialization argument

Reply:
Agree, and I think it should be supported in `config_list` since weight and activation may use different settings and even different layers can use different settings.

Reply:
`dtype` and `qscheme` can be applied on each layer separately. Also we can customize `validate_config` function to control the rules for each quantizer